gammapy/datasets/make.py

```diff
@@ -209,3 +211,128 @@ def make_test_observation_table(observatory_name, n_obs, debug=False):
+        minimum energy
+    energy_max : `~astropy.units.Quantity`, optional
+        maximum energy
+    nenergy_bins : int, optional
+        number of (logarithmic) bins in energy
+    sigma : `~astropy.coordinates.Angle`, optional
+        width of the gaussian model used for the spatial coordinates
+    index : double, optional
+        index for the power-law model used for the energy coordinate
+    apply_mask : bool, optional
+        if set, 1/4th of the image is masked (for `x > x_center` and
+        `y > y_center`)
+
+    Returns
+    -------
+    bg_cube_model : `~gammapy.background.CubeBackgroundModel`
+        bg cube model
+    """
+    # spatial bins (linear)
+    delta_x = (detx_max - detx_min)/ndetx_bins
+    detx_bin_edges = np.arange(ndetx_bins + 1)*delta_x + detx_min
+
+    delta_y = (dety_max - dety_min)/ndety_bins
+    dety_bin_edges = np.arange(ndety_bins + 1)*delta_y + dety_min
+
+    # energy bins (logarithmic)
+    log_delta_energy = (np.log(energy_max.value)
+                        - np.log(energy_min.value))/nenergy_bins
+    energy_bin_edges = np.exp(np.arange(nenergy_bins + 1)*log_delta_energy
+                              + np.log(energy_min.value))
+    energy_bin_edges = Quantity(energy_bin_edges, energy_min.unit)
+    # TODO: this function should be reviewed/re-written, when
+    # the following PR is completed:
+    # https://github.com/gammapy/gammapy/pull/290
+
+    # define empty bg cube model and set bins
+    bg_cube_model = CubeBackgroundModel(detx_bins=detx_bin_edges,
+                                        dety_bins=dety_bin_edges,
+                                        energy_bins=energy_bin_edges,
+                                        background=None)
+
+    # background
+
+    # define coordinate grids for the calculations
+    det_bin_centers = bg_cube_model.image_bin_centers
+    energy_bin_centers = bg_cube_model.energy_bin_centers
+    energy_points, dety_points, detx_points = np.meshgrid(energy_bin_centers,
+                                                          det_bin_centers[1],
+                                                          det_bin_centers[0],
+                                                          indexing='ij')
+    E_0 = Quantity(1., 'TeV')
+    norm = Quantity(1., '1 / (s TeV sr)')
+
+    # define E dependent sigma
+    # it is defined via a PL, in order to be log-linear
+    # it is equal to the parameter sigma at E max
+    # and sigma/2. at E min
+    sigma_min = sigma/2. # at E min
+    sigma_max = sigma # at E max
+    s_index = np.log(sigma_max/sigma_min)
+    s_index /= np.log(energy_bin_edges[-1]/energy_bin_edges[0])
+    s_norm = sigma_min*((energy_bin_edges[0]/E_0)**-s_index)
+    sigma = s_norm*((energy_points/E_0)**s_index)
+
+    # calculate bg
+    gaussian = np.exp(-((detx_points)**2 + (dety_points)**2)/sigma**2)
+    powerlaw = (energy_points/E_0)**-index
+    background = norm*gaussian*powerlaw
+
+    # apply mask if requested
```

Comment:
IMO numpy masked arrays cause trouble (performance issues, unexpected behaviour with a few numpy functions) sooner or later.
My suggestion would be to filling zeros here instead of making a masked array!?